### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Rombur**
This PR updates how to install and run the code. I've also added comments every time we get a value from a property tree. This should make it easier to maintain the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,165 @@
...
+Then simply use `make`. This will compile `adamantine` and create an executable
+in a newly created `bin` subdirectory. You will find in this subdirectory the
+executable and an example of input files.
+
+The list of configuration options is:
+* CMAKE\_BUILD\_TYPE: Debug/Release
+* DEAL\_II\_DIR:/path/to/dealii 
+* BOOST\_DIR=/path/to/boost
+* ADAMANTINE\_ENABLE\_TESTS=ON/OFF
+* ADAMANTINE\_ENABLE\_COVERAGE=ON/OFF
```
Comment (stvdwtt) [inline README.md:34]: There should also be an entry for compiling with CUDA support, right?
`-D ADAMANTINE_ENABLE_CUDA=ON `

File: README.md [hunk 2]
```diff
@@ -1,10 +1,165 @@
...
+  thermal\_conductivity (optional)
+  * material\_X.A: A is either solidus, liquidus, or latent\_heat (optional)
+* sources:
+  * n\_beams: number of electron beams
+  * beam\_X: property tree for the beam with number X
+  * beam\_X.type: type of heat source: goldak or electron\_beam 
+  * beam\_X.scan\_path\_file: scan path filename
+  * beam\_X.scan\_path\_file\_format: format of the scan path: segment or
+  event\_series
+  * beam\_X.depth: maximum depth reached by the electron beam in meter
```
Comment (stvdwtt) [inline README.md:88]: typo -- should be "meters"

File: README.md [hunk 3]
```diff
@@ -1,10 +1,165 @@
...
+  * beam\_X.absorption\_efficiency: absorption efficiency of the beam equivalent
+  to energy\_conversion\_efficiency * control\_efficiency for electon beam. Number
+  between 0 and 1.
+  * beam\_X.diameter: diameter of the beam in meter (default value: 2e-3)
```
Comment (stvdwtt) [inline README.md:92]: typo -- should be "meters"

File: README.md [hunk 4]
```diff
@@ -1,10 +1,165 @@
...
+* materials:
+  * n\_materials: number of materials
+  * property\_format: format of the material property: table or polynomial
+  * initial\_temperature: initial temperature of all the materials (default value: 300)
+  * material\_X: property tree for the material with number X
+  * material\_X.Y: property tree where Y is either liquid, powder, or solid
+  (optional)
+  * material\_X.Y.Z: Z is either density, specific\_heat, or
+  thermal\_conductivity (optional)
+  * material\_X.A: A is either solidus, liquidus, or latent\_heat (optional)
```
Comment (stvdwtt) [inline README.md:80]: What are the units?

File: README.md [hunk 5]
```diff
@@ -1,10 +1,165 @@
...
+  * beam\_X.absorption\_efficiency: absorption efficiency of the beam equivalent
+  to energy\_conversion\_efficiency * control\_efficiency for electon beam. Number
+  between 0 and 1.
+  * beam\_X.diameter: diameter of the beam in meter (default value: 2e-3)
+* time\_stepping:
+  * method: name of the method to use for the time integration: forward\_euler,
+  rk\_third\_order, rk\_fourth\_order, heun\_euler, bogacki\_shampine, dopri,
+  fehlberg, cash\_karp, backward\_euler, implicit\_midpoint, crank\_nicolson, or
+  sdirk2
+  * duration: duration of the simulation in second
```
Comment (stvdwtt) [inline README.md:98]: typo -- should be "seconds"

File: README.md [hunk 6]
```diff
@@ -1,10 +1,165 @@
...
+  * time\_step: length of the time steps used for the simulation in second
```
Comment (stvdwtt) [inline README.md:99]: typo -- should be "seconds"

File: README.md [hunk 7]
```diff
@@ -1,10 +1,165 @@
...
+    * tolerance: tolerance of the linear solver (default value: 1e-12)
+    * n\_tmp\_vectors: maximum number of vectors used by GMRES (default value:
+    30)
+    * right\_preconditioner: use left or right preconditioning for the linear
+    solver (default value: false)
+    * newton\_max\_iteration: maximum number of iterations of Newton solver
+    (default value: 100)
+    * newton\_tolerance: tolerance of the Newton solver (default value: 1e-6)
+    * jfnk: use Jacobian-Free Newton Krylov method (default value: false)
+    * right\_preconditioner: use left or right precondioning for the linear solver
```
Comment (stvdwtt) [inline README.md:123]: Repeat from line 117

File: README.md [hunk 8]
```diff
@@ -1,10 +1,165 @@
...
+    * right\_preconditioner: use left or right precondioning for the linear solver 
+    (default value: false)
+* post\_processor:
+  * file\_name: prefix of output files
+* discretization:
+  * fe\_degree: degree of the finite element used
+  * quadrature: quadrature used: gauss or lobatto (default value: gauss)
+* profiling (optional):
+  * timer: output timing information (default value: false)
+* memory\_space: device (use GPU) or host (use CPU) (default value: host)
```
Comment (stvdwtt) [inline README.md:132]: Is this a way to change between using the GPU vs CPU-only at runtime? I'm more familiar with Kokkos than CUDA -- is it assumed that the execution space matches the memory space here?

File: README.md [hunk 9]
```diff
@@ -1,10 +1,165 @@
...
+## Run
+After compiling `adamantine`, you can run a simulation using
+```bash
+mpirun -n 2 ./adamantine --input-file=input.info
+```
+Note that the name of the input file is totally arbitrary, `my_input_file` is as
+valid as `input.info`.
+
+### Input file
+The following options are available:
```
Comment (stvdwtt) [inline README.md:45]: I didn't go through the source files to make sure that this is exhaustive. Let me know if you'd like me to. It _looks_ like everything is listed.

File: README.md (hunk 8 above)
Comment (Rombur) [reply README.md:132]: > Is this a way to change between using the GPU vs CPU-only at runtime?

Yes, exactly. 

> is it assumed that the execution space matches the memory space here?

You could say that. In Kokkos, you can have multiple execution spaces for a memory space. For example for the Host memory space, the execution space can be Serial or OpenMP. Here you have no such choice. On the host, the execution space is serial and on the device the execution space is Cuda.

File: README.md (hunk 9 above)
Comment (Rombur) [reply README.md:45]: No, don't worry. I don't want you to go through the code.

**Comment by Rombur** [comment]
This is ready to be merged

**Review by stvdwtt — APPROVED** [review]

**Pull request merged**